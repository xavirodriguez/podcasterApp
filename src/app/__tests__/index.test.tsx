import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import { App } from '../index';

const renderer = createRenderer();

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));
describe('<App />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<App />);
    const renderedOutput = renderer.getRenderOutput();

    expect(renderedOutput).toMatchSnapshot();
  });
});
