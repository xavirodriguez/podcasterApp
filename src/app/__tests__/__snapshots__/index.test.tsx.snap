// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<BrowserRouter>
  <Helmet
    defaultTitle="Podcaster"
    defer={true}
    encodeSpecialCharacters={true}
    htmlAttributes={
      Object {
        "lang": undefined,
      }
    }
    prioritizeSeoTags={false}
    titleTemplate="Podcaster"
  >
    <meta
      content="Podcaster App"
      name="description"
    />
  </Helmet>
  <PageHeader />
  <Routes>
    <Route
      element={<Unknown />}
      path="/"
    />
    <Route
      element={<PodcastPage />}
      path="/podcast/:id"
    />
    <Route
      element={<EpisodePage />}
      path="/podcast/:id/episode/:trackId"
    />
    <Route
      element={<Unknown />}
      path="*"
    />
  </Routes>
  <Memo(l) />
</BrowserRouter>
`;
